pkg/agent/controller/traceflow/packetin.go

```diff
@@ -176,10 +180,22 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 				return nil, nil, err
 			}
 		}
-		if tunnelDstIP != "" && tunnelDstIP != c.nodeConfig.NodeIPAddr.IP.String() {
-			ob.TunnelDstIP = tunnelDstIP
+		var outputPort uint32
+		if match = getMatchRegField(matchers, uint32(openflow.PortCacheReg)); match != nil {
+			outputPort, err = getInfoInReg(match, nil)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+		if outputPort == config.DefaultTunOFPort || outputPort == config.HostGatewayOFPort {
+			// Output port is Tunnel/Gateway port, packet is forwarded
+			if tunnelDstIP != "" {
```

Comment:
Nit: you can also skip the check, and always set ob.TunnelDstIP to tunnelDstIP.

Reply:
Removed this check